src/Components/Server/src/Builder/ServerRazorComponentsEndpointConventionBuilderExtensions.cs

```diff
@@ -54,7 +54,20 @@ public static RazorComponentsEndpointConventionBuilder AddInteractiveServerRende
                         var original = b.RequestDelegate;
                         b.RequestDelegate = async context =>
                         {
-                            context.Response.Headers.Add("Content-Security-Policy", $"frame-ancestors {options.ContentSecurityFrameAncestorsPolicy}");
+                            if (context.Response.Headers.ContentSecurityPolicy.Count == 0)
+                            {
+                                context.Response.Headers.ContentSecurityPolicy = $"frame-ancestors {options.ContentSecurityFrameAncestorsPolicy}";
+                            }
+                            else
+                            {
+                                var result = new string[context.Response.Headers.ContentSecurityPolicy.Count + 1];
+                                for (var i = 0; i < result.Length - 1; i++)
+                                {
+                                    result[i] = context.Response.Headers.ContentSecurityPolicy[i];
+                                }
+                                result[^1] = $"frame-ancestors {options.ContentSecurityFrameAncestorsPolicy}";
+                                context.Response.Headers.ContentSecurityPolicy = result;
+                            }
```

Comment:
Would the following work as a simplification? I think it's equivalent. The `StringValues.Concat` static method already does almost exactly what you're doing manually here.

```suggestion
                            var headers = context.Response.Headers;
                            headers.ContentSecurityPolicy = StringValues.Concat(headers.ContentSecurityPolicy, $"frame-ancestors {options.ContentSecurityFrameAncestorsPolicy}");
```

Reply:
Wasn't aware of StringValues concat.